qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -417,22 +423,37 @@ def __init__(self, name: str, address: str, init_s_params: bool=True, **kwargs)
                            get_cmd='OUTP1?',
                            set_cmd='OUTP1 {}',
                            val_mapping={True: '1\n', False: '0\n'})
-        self.add_function('reset', call_cmd='*RST')
-        self.add_function('tooltip_on', call_cmd='SYST:ERR:DISP ON')
-        self.add_function('tooltip_off', call_cmd='SYST:ERR:DISP OFF')
-        self.add_function('cont_meas_on', call_cmd='INIT:CONT:ALL ON')
-        self.add_function('cont_meas_off', call_cmd='INIT:CONT:ALL OFF')
-        self.add_function('update_display_once', call_cmd='SYST:DISP:UPD ONCE')
-        self.add_function('update_display_on', call_cmd='SYST:DISP:UPD ON')
-        self.add_function('update_display_off', call_cmd='SYST:DISP:UPD OFF')
-        self.add_function('display_sij_split', call_cmd='DISP:LAY GRID;:DISP:LAY:GRID {},{}'.format(
-            num_ports, num_ports))
-        self.add_function('display_single_window',
-                          call_cmd='DISP:LAY GRID;:DISP:LAY:GRID 1,1')
-        self.add_function('display_dual_window',
-                          call_cmd='DISP:LAY GRID;:DISP:LAY:GRID 2,1')
-        self.add_function('rf_off', call_cmd='OUTP1 OFF')
-        self.add_function('rf_on', call_cmd='OUTP1 ON')
+
+        self.add_parameter(name='tooltip',
+                           set_cmd='SYSTem:ERRor:DISPlay {}',
+                           get_cmd='SYSTem:ERRor:DISPlay?',
+                           get_parser=int,
+                           val_mapping={"ON": 1, "OFF": 0})
```

Comment:
do i understand correctly that here both numbers and strings work for this SCPI command?

Reply:
I think I'd rather say that both strings and integers work with python's string formatter.

It goes like this: The user inputs `"ON"`. The `val_mapping` converts that into `1`. That value is passed to the `set_cmd` that calls `.format(1)`. The result, `'SYSTem:ERRor:DISPlay 1'` is sent via VISA to the instrument. 

Now, upon `get`, the string `"1\n"` is obtained. It goes through the `get_parser` and becomes `1`. The `val_mapping` then turns that into `"ON"` and the cycle is complete.